### ISSUE TITLE
Memoising tidygeocoder::geo leads to "promise already under evaluation: recursive default argument reference or earlier problems?"

### ISSUE THREAD

**Issue by dewoller** [r-lib/memoise#131]
The following code leads to the above error:

```
library(tidygeocoder)
library(memoise)
mf_geo = memoise( geo )
mf_geo( address= "IGA WEST DUBBO VICTORIA STREET WEST DUBBO NSW Australia"  )
```

`promise already under evaluation: recursive default argument reference or earlier problems?`

The unmemoised version works:

`geo( address= "IGA WEST DUBBO VICTORIA STREET WEST DUBBO NSW Australia" )`

**Comment by jimhester** [comment]
The issue is the `geo` function has `lat = lat, long = long` and you can't have a default argument with a default value that is the same as the argument name.

``` r
f <- function(x, y = y) {
  force(y)
  x
}
f()
#> Error in force(y): promise already under evaluation: recursive default argument reference or earlier problems?
```

<sup>Created on 2021-08-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

The original function must not ever force the promises with these default arguments? Otherwise you would see the same error with `geo` as you do with `mf_geo`.

memoise forces all arguments first so it can include their values in the hash.

**Comment by dewoller** [comment]
I think my proper response is "Duh!".  Now the error message makes sense also.  Thanks for the great package, very handy.

**Comment by dpprdan** [comment]
This should be fixed in version 1.0.5, now on CRAN. See https://github.com/jessecambon/tidygeocoder/pull/154. 

Thanks for reporting, @dewoller, and thanks for debugging, @jimhester!
